src/main/java/com/github/invictum/panels/PanelFactory.java

```diff
@@ -53,18 +58,53 @@ public static void setPanelInitWaitStrategy(PanelInitStrategy strategyToSet) {
      * @return Panel
      */
     public static <T extends AbstractPanel> T get(final Class<T> panelClass, final AbstractPage parentPage) {
+        applyGlobalStrategy(panelClass, parentPage);
         T panelInstance = getPanel(panelClass);
-        if (!panelClass.isAnnotationPresent(DisableGlobalInitStrategy.class)) {
-            strategy.apply(parentPage);
+        String locator = UnifiedDataProviderFactory.getInstance(panelInstance).getBase();
+        WebElementFacade panel;
+        if (locator == null) {
+            panel = parentPage.find(By.xpath(FLOATING_PANEL_BASE_LOCATOR));
+        } else {
+            panel = parentPage.isXpath(locator) ? parentPage.find(By.xpath(locator)) : parentPage
+                    .find(By.cssSelector(locator));
         }
-        panelInstance.initWith(parentPage);
+        /** Wrap panel element into panel class. */
+        panelInstance.initWith(parentPage, panel);
         invokeWhenInitializedMethods(panelInstance);
         return panelInstance;
     }
 
+    public static <T extends AbstractPanel> List<T> getAll(final Class<T> panelClass, final AbstractPage parentPage) {
+        applyGlobalStrategy(panelClass, parentPage);
+        String locator = UnifiedDataProviderFactory.getInstance(getPanel(panelClass)).getBase();
+        List<WebElementFacade> panels;
+        if (locator == null) {
+            panels = parentPage.findAll(By.xpath(FLOATING_PANEL_BASE_LOCATOR));
```

Comment:
As for me there no need to implement .getAll() method for panels without locator. It's really bad practice to init list of all body. So I can propose 2 solution:
1. Implement .getAll() method inside of AbstractPanel class.
2. Throw exception for null locator with reasonable error.